### PR TITLE
Repo Revival 

### PR DESCRIPTION
**Pull request by salmad3**
With the amount of docs, we should update the structure of repo sooner then later.  

This will attempt to address suggestion(s) in https://github.com/w3f/polkadot-wiki/issues/2294, and more recently, https://github.com/w3f/polkadot-wiki/issues/2415#issuecomment-892019159, but if we do decide to take the address convertor out, restructure could still be useful [wondering what the best approach would be]

- v1 static files can live somewhere else.

Some thing I noticed when updating the links: I think when referencing a link, having the link flow as a sentence and limiting the reference from stating the link as a location is better suited. So, **not** using `here`: e.g. "...which is available [here](link)")

Added a pretty check that runs automatically pre-commit

Also, does anyone know why `build-ss58-registry.md` autogenerates at each run (the chosen one)? (package.json: `    "generateRegistryTable": "node --experimental-json-modules scripts/generateRegistryTable.mjs > docs/build-ss58-registry.md; npx prettier docs/build-ss58-registry.md --write",`

### CODE REVIEW

**Comment by DannyS03** [comment]
To restructure the js, I'm wondering how we can approach it since we basically have two webpages in separate directories for each wiki

File: docs/learn/learn-staking.md [hunk 1]
```diff
@@ -67,16 +68,23 @@ with stake `s_i`. In that case, nominator `n` will be rewarded essentially the s
 For each validator pool, we keep a list of nominators with the associated stakes.
 
 The general rule for rewards across validator pools is that two validator pools get paid essentially
+<<<<<<< HEAD:docs/learn/learn-staking.md
```
Comment (emresurmeli) [inline docs/learn/learn-staking.md]: Looks like some merge conf left overs

File: docs/learn/learn-staking.md [hunk 2]
```diff
@@ -194,7 +202,8 @@ responsibilities regarding data availability and [XCMP](learn-crosschain).
 Nominators, on the other hand, have far fewer responsibilities. Those include monitoring their
 validators' performance (uptime), keeping an eye on changing commission rates (a validator can
 change commission at any time), and general health monitoring of their and their validators'
-account. Thus, while not exactly set-it-and-forget-it, a nominator's experience is pretty hands-off.
+account. Thus, while not set-it-and-forget-it, a nominator's experience is relatively hands-off
+compared to a validator's.
```
Comment (emresurmeli) [inline docs/learn/learn-staking.md]: validators*

File: docs/learn/learn-staking.md [hunk 3]
```diff
@@ -279,17 +290,27 @@ offline during the epoch, slashing may occur.
...
 
     min((3 * (x - (n / 10 + 1))) / n, 1) * 0.07
 
+Let us run through a few examples to understand this equation. In all of the examples, assume that
+there are 100 validators in the active set.
+
 Note that if less than 10% of all validators are offline, no penalty is enacted.
 
 Validators should have a well-architected network infrastructure to ensure the node is running to
+<<<<<<< HEAD:docs/learn/learn-staking.md
```
Comment (emresurmeli) [inline docs/learn/learn-staking.md]: Git merge conf left overs

File: docs/learn/learn-staking.md [hunk 4]
```diff
@@ -279,17 +290,27 @@ offline during the epoch, slashing may occur.
...
 reduce the risk of being slashed. A high availability setup is desirable, preferably with backup
 nodes that kick in **only once the original node is verifiably offline** (to avoid double-signing
 and being slashed for equivocation - see below). A comprehensive guide on validator setup is
 available [here](../maintain/maintain-guides-secure-validator.md).
+=======
```
Comment (emresurmeli) [inline docs/learn/learn-staking.md]: ++

**Comment by jakehemmerle** [comment]
This PR is way too big to review all at once. At the very least, the folder relocation needs to be split into a separate PR. Ideally one commit for the folder relocations, another where all the references are changed. Noone can go through 1250 files file.

**Comment by salmad3** [comment]
> This PR is way too big to review all at once. At the very least, the folder relocation needs to be split into a separate PR. Ideally one commit for the folder relocations, another where all the references are changed. Noone can go through 1250 files file.

For sure haha

**Comment by salmad3** [comment]
@jakehemmerle The issue I'm finding with splitting it up is the updates are dependant on eachother, so the site wouldn't build properly and CI wouldn't pass - the changes in the files are mainly changes in paths/links and they are "re-added" since their location changes - I'd hope running the sites locally will point out if there is something wrong

Changes to point out
- new directories for each set of docs: general, learn, build, maintain (kusama specific docs are under their own folder, e.g. general/kusama): any broken links or path references 
- mirror.mjs (can check that mirroring works)
- v1 has been taken out
- some files taken out 
- paths in js have been updated (e.g. index.js)
- added `pretty-quick` in substitute of standard prettier: there is also general reformatting updates after running this

File: docs/learn/learn-staking.md (hunk 1 above)
Comment (salmad3) [reply docs/learn/learn-staking.md]: Thanks!

**Comment by DrW3RK** [comment]
As most of this is done manually, I'm good with merging this with Master branch. We can thoroughly test the staging website before rebasing prod

**Review by DrW3RK — APPROVED** [review]

**Pull request merged**